client/proxguiqt.cpp

```diff
@@ -685,6 +689,26 @@ void Plot::keyPressEvent(QKeyEvent *event) {
             master->hide();
             break;
 
+        case Qt::Key_Home:
+            GraphStart = 0;
+            break;
+
+        case Qt::Key_End:
+            GraphStart = startMax;
+            break;
+
+        case Qt::Key_PageUp:
+            if (GraphStart >= PageWidth) {
+                GraphStart -= PageWidth;
+            } else {
+                GraphStart = 0;
+            }
+            break;
+
+        case Qt::Key_PageDown:
+            GraphStart += PageWidth;
```

Comment:
No check for larger than startMax?  ie overshoot number of samples available

Reply:
It gets clamped in the paint function so I never bothered. I can easily check though, I'll throw it in tomorrow.

Thanks!

Reply:
do another PR with that fix.  I merge this one meanwhile.